geometry/create-and-edit-geometries/src/main/java/com/esri/samples/create_and_edit_geometries/CreateAndEditGeometriesController.java

```diff
@@ -0,0 +1,591 @@
+
+    // set the map to the map view
+    mapView.setMap(map);
+
+    // set a viewpoint on the map view
+    mapView.setViewpoint(new Viewpoint(53.08230, -9.5920, 5000));
+
+    // create a graphics overlay for the graphics
+    graphicsOverlay = new GraphicsOverlay();
+
+    // add the graphics overlay to the map view
+    mapView.getGraphicsOverlays().add(graphicsOverlay);
+
+    // create a new sketch editor and add it to the map view
+    geometryEditor = new GeometryEditor();
+    mapView.setGeometryEditor(geometryEditor);
+
+    // create vertex and freehand tools for the geometry editor and add to combo box - vertex tool selected by default
+    vertexTool = new VertexTool();
+    freehandTool = new FreehandTool();
+    toolComboBox.setConverter(new ComboBoxStringConverter());
+    toolComboBox.getItems().addAll(vertexTool, freehandTool);
+    toolComboBox.getSelectionModel().select(0);
+
+    // add example graphics to the map
+    createGraphicsMarkers();
+    createInitialGraphics();
+
+    // bind button availability to properties of the geometry editor
+    bindButtonsToGeometryEditor();
+
+    // allow user to immediately select and edit graphics
+    selectGraphic();
+  }
+
+  /**
+   * Binds button status to geometry editor properties.
+   */
+  private void bindButtonsToGeometryEditor() {
+    // point and multipoint disabled when freehand tool selected
+    pointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+    multipointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+
+    // only show the selected geometry type as enabled while editing
+    geometryEditor.startedProperty().addListener((ob, ov, nv) -> {
+      if (geometryEditor.isStarted()) {
+        pointButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.POINT)));
+        multipointButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.MULTIPOINT)));
+        polylineButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.POLYLINE)));
+        polygonButton.disableProperty().bind(new SimpleBooleanProperty(!geometryEditor.getGeometry().getGeometryType()
+          .equals(GeometryType.POLYGON)));
+      } else {
+        // when editor stopped, re-enable polygon & polyline, enable point/multipoint if vertex tool selected
+        pointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+        multipointButton.disableProperty().bind(geometryEditor.toolProperty().isEqualTo(freehandTool));
+        polylineButton.disableProperty().bind(new SimpleBooleanProperty(false));
+        polygonButton.disableProperty().bind(new SimpleBooleanProperty(false));
+      }
+    });
```

Comment:
```suggestion
    geometryEditor.startedProperty().addListener((observable, oldValue, newValue) -> {
      if (newValue) {
        // when the geometry editor is started (and so the geometry will not be null) bind the button state to the geometry type
        pointButton.disableProperty().bind(Bindings.createBooleanBinding(() -> geometryEditor.getGeometry().getGeometryType().equals(GeometryType.POINT)).not());
        multipointButton.disableProperty().bind(Bindings.createBooleanBinding(() -> geometryEditor.getGeometry().getGeometryType().equals(GeometryType.MULTIPOINT)).not());
        polylineButton.disableProperty().bind(Bindings.createBooleanBinding(() -> geometryEditor.getGeometry().getGeometryType().equals(GeometryType.POLYLINE)).not());
        polygonButton.disableProperty().bind(Bindings.createBooleanBinding(() -> geometryEditor.getGeometry().getGeometryType().equals(GeometryType.POLYGON)).not());
        toolComboBox.disableProperty().bind(Bindings.or(
            Bindings.createBooleanBinding(() -> geometryEditor.getGeometry().getGeometryType().equals(GeometryType.MULTIPOINT)),
            Bindings.createBooleanBinding(() -> geometryEditor.getGeometry().getGeometryType().equals(GeometryType.POINT))
        ));
      } else {
        // when the geometry editor is not started (and so the geometry is null) enable all geometry buttons
        pointButton.disableProperty().unbind();
        pointButton.setDisable(false);
        multipointButton.disableProperty().unbind();
        multipointButton.setDisable(false);
        polylineButton.disableProperty().unbind();
        polylineButton.setDisable(false);
        polygonButton.disableProperty().unbind();
        polygonButton.setDisable(false);
        toolComboBox.disableProperty().unbind();
        toolComboBox.setDisable(true);
      }
    });
```
Functionally this suggestion is more or less the same, just using the Bindings API rather than creating properties, and also you can see here the combox controls too. In addition, we can remove the binding and set the state (rather than creating a "mock" property). As I'm sure you encountered, if we don't "unbind" here, then we'd get an error that a "bound value cannot be set".

Reply:
Also note that we probably want to revisit some of the documentation/comments, feel free to edit as you see fit.